src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/FileEntryCreator.kt

```diff
@@ -0,0 +1,78 @@
+package pl.edu.uj.ii.ksi.mordor.services
+
+import java.io.File
+import org.springframework.stereotype.Service
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileContent
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileEntry
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileMetadata
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileContentRepository
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileEntryRepository
+import pl.edu.uj.ii.ksi.mordor.persistence.repositories.FileMetadataRepository
+import pl.edu.uj.ii.ksi.mordor.services.hash.FileHashProvider
+
+@Service
+class FileEntryCreator(
+    private val metadataExtractor: MetadataExtractor,
+    private val entryRepository: FileEntryRepository,
+    private val metadataRepository: FileMetadataRepository,
+    private val contentRepository: FileContentRepository,
+    private val fileTextExtractor: FileTextExtractor,
+    private val hashProvider: FileHashProvider
+) {
+
+    companion object {
+        // TODO: move to config
+        private const val contentMaxLength: Int = 24 * 1024
+    }
+
+    fun create(file: File) {
```

Comment:
Creating a shitton of tiny functions does NOT actually improve readability. Please apply common sense.